### ISSUE TITLE
Lookup Tool Stalls

### ISSUE THREAD

**Issue by jaickler** [brianpetro/obsidian-smart-connections#917]
# Bug
When using smart chat the lookup tool will show hypotheticals and then the loading wheel will display indefinitely. 
# Expected behavior
Recurve message from LLM after lookup finishes. 
# How to replicate
Use gpt-4o-mini with tools results being sent in user message or not. 
Trigger the lookup tool. 
Wait indefinitely for message to return. 
# Notes
I am using Linux on my laptop and do not have mobile enabled. Embedding have fully loaded already.

**Comment by mnmnk43434** [comment]
Hello, we tried to solve the issue.

This is what we did:

Modified the `get_context_hyde` function in `sc_actions.js` to ensure the chat continues after the lookup tool completes. Added error handling and made sure the chat model is called again after the lookup results are added.

You can review changes in this commit: https://github.com/mnmnk43434/brianpetro-obsidian-smart-connections-917/commit/1aee4237152cf06ad501aed5f4abc39f7eda9f84.


> [!CAUTION]
> Disclaimer: The concept of solution was created by AI and you should never copy paste this code before you check the correctness of generated code. Solution might not be complete, you should use this code as an inspiration only.

---

If you no longer want Latta AI to attempt solving issues on your repository, you can block this account.

**Comment by brianpetro** [comment]
Hi @jaickler 

Are there any errors in the developer console logs when this happens?

Thanks for the report
🌴

**Comment by jaickler** [comment]
Hello! Yes, I was able to check the developer logs and saw the vector sizes didn't match. I deleted the multi folder and then relaunched Obsidian and that fixed half the issue. The other half is that when using OpenAI's 4o-mini model I have to send the tool results in the user message or I also receive errors. 
Sending the results in the user message does allow chat to work normally though.

I suppose the nature of the current bug is that, at least for 4o-mini the tool results must be sent in the user prompt.

**Comment by brianpetro** [comment]
Thanks for following up. 

4o-mini seems to be working with/without sending tools response in the user message. Does toggling that setting on/off consistently produce the same success/fail results for you?

Please do let me know if you notice any other patterns or anything that might help narrow down what might be causing the issue 🌴

**Comment by Keffus** [comment]
I'm having the same problem with gemini

**Comment by brianpetro** [comment]
@jaickler @Keffus any screenshots that might help me better see what's going on would be helpful 🌴

**Comment by Keffus** [comment]
yeah sure, on the first screenshot i asked gemini if it can summarize the wiki description of cat and it did it but only in lookup, and it was loading the answer indefinitely. But when i asked the next question it responded normally(second screenshot). I think the lookup thing is related to asking about the vault because when i dont do it it answers normally. I have the review context option turned off.
![Capture1](https://github.com/user-attachments/assets/21d54446-f091-46b7-b554-ad48034eea1d)
![Capture2](https://github.com/user-attachments/assets/451128b1-c3c0-485b-a9f8-4094b73a5035)

**Comment by Keffus** [comment]
I check one more think i opened a new chat and just asked if it can tell me somethink(didn't ask to use vault) and it did answer in lookup only so I don't think it is realed to vault

**Comment by Keffus** [comment]
I don't know how much this will help but i have figure out that if gemini can't answer the question it will anser normally(without lookup
![Capture1](https://github.com/user-attachments/assets/302e1edb-3dd3-4375-a4c3-7b5331fee8e0)
![Capture2](https://github.com/user-attachments/assets/fab626aa-8485-44eb-84ee-0a2f92807017)
)
